src/snowflake/connector/result_set.py

```diff
@@ -178,19 +178,23 @@ def _fetch_pandas_batches(self, **kwargs) -> Iterator[DataFrame]:
         Snowflake's back-end.
         """
         self._can_create_arrow_iter()
-        return self._create_iter(iter_unit=IterUnit.TABLE_UNIT, structure="pandas")
+        return self._create_iter(
+            iter_unit=IterUnit.TABLE_UNIT, structure="pandas", **kwargs
+        )
 
     def _fetch_pandas_all(self, **kwargs) -> DataFrame:
         """Fetches a single Pandas dataframe."""
-        dataframes = list(self._fetch_pandas_batches())
+        dataframes = list(self._fetch_pandas_batches(**kwargs))
         if dataframes:
+            concat_kwargs = kwargs
+            concat_kwargs.pop("types_mapper")
             return pandas.concat(
                 dataframes,
                 ignore_index=True,  # Don't keep in result batch indexes
-                **kwargs,
```

Comment:
You should be able to pass in **kwargs because non-used arguments will be ignored

Reply:
It will throw a VauleError unfortunately.